files/fluentd/run.sh

```diff
@@ -68,8 +67,31 @@ fi
 # Determine allowed total given the number of outputs we have.
 ALLOWED_DF_LIMIT=$(expr $DF_LIMIT \* $ALLOWED_PERCENT_OF_DISK / 100) || :
 
+TOTAL_LIMIT_SIZE_CONF=$(grep "total_limit_size" $FLUENT_CONF |  head -1 | awk '{print $2}' )
+
+if [[ $TOTAL_LIMIT_SIZE_CONF == *"TOTAL_LIMIT_SIZE"* ]]; then
+   TOTAL_LIMIT_SIZE_CONF=''
+else
+   TOTAL_LIMIT_SIZE_CONF=$(echo $TOTAL_LIMIT_SIZE_CONF |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc) || :
+fi
+
+TOTAL_LIMIT_SIZE_CONF=${TOTAL_LIMIT_SIZE_CONF:-0}
+
 # TOTAL_LIMIT_SIZE per buffer
-TOTAL_LIMIT_SIZE=$(expr $ALLOWED_DF_LIMIT / $NUM_OUTPUTS) || :
+TOTAL_LIMIT_SIZE_ALLOWED=$(expr $ALLOWED_DF_LIMIT / $NUM_OUTPUTS) || :
+TOTAL_LIMIT_SIZE_ALLOWED=${TOTAL_LIMIT_SIZE_ALLOWED:-0}
+
+if [[ $TOTAL_LIMIT_SIZE_CONF -lt $TOTAL_LIMIT_SIZE_ALLOWED ]];
+then
+   if [[ $TOTAL_LIMIT_SIZE_CONF -gt 0 ]]; then
+       TOTAL_LIMIT_SIZE=$TOTAL_LIMIT_SIZE_CONF
+   else
+       TOTAL_LIMIT_SIZE=$TOTAL_LIMIT_SIZE_ALLOWED
+   fi
+else
+    TOTAL_LIMIT_SIZE=$TOTAL_LIMIT_SIZE_ALLOWED
```

Comment:
lets echo a message why we are disregarding... "Requested exceeds the maximum available...." or something like that